src/components/v2v3/shared/FundingCycleConfigurationDrawers/NftDrawer/MaxSupplyFormItem.tsx

```diff
@@ -1,20 +1,18 @@
 import { t } from '@lingui/macro'
 import { Form, Switch } from 'antd'
+import { FormInstance, useWatch } from 'antd/lib/form/Form'
 import FormattedNumberInput from 'components/inputs/FormattedNumberInput'
 import TooltipLabel from 'components/TooltipLabel'
 
-import { useState } from 'react'
+import { NftFormFields } from './NftRewardTierModal'
 
 export default function MaxSupplyFormItem({
-  value,
-  onChange,
+  form,
 }: {
-  value: number | undefined
-  onChange: (value: number | undefined) => void
+  form: FormInstance<NftFormFields>
 }) {
-  const [limitedSupplyEnabled, setLimitedSupplyEnabled] = useState<boolean>(
-    Boolean(value),
-  )
+  const value = useWatch('maxSupply', form)
```

Comment:
Recommend using form instance directly rather than passing it in:

```
const form = Form.useFormInstance()
...
```

Saves the need to prop it in

Reply:
Ah nice didn't know about this, thank you sir.